mpqp/core/instruction/measurement/expectation_value.py

```diff
@@ -204,16 +204,20 @@ def __init__(
         super().__init__(targets, shots, label)
         self.observable = observable
         """See parameter description."""
-        # Raise an error if the number of target qubits does not match the size of the observable.
+
+        if targets is None:
+            self.pre_measure = QCircuit(0)
+            return
+
         if self.nb_qubits != observable.nb_qubits:
             raise NumberQubitsError(
-                f"{self.nb_qubits}, the number of target qubit(s) doesn't match"
-                f" {observable.nb_qubits}, the size of the observable"
+                f"Target size {len(targets)} doesn't match observable size "
```

Comment:
Why do we use `len(targets)` instead of `self.nb_qubits` ? We already use the second in the `if`, and it would avoid calling `len` again